blocks/monitoring/aggregation.py

```diff
@@ -151,3 +151,19 @@ def get_aggregator(self):
                               (self.storage, tensor.zeros_like(self.storage))],
                           accumulation_updates=[(self.storage, self.variable)],
                           readout_variable=self.storage)
+
+
+@add_metaclass(ABCMeta)
+class MonitoredQuantity(object):
+    def __init__(self, requires=None, name=None):
+        self.requires = requires
+        self.name = name
+        pass
+
```

Comment:
Sorry, reseting the counters in the `readout` method is not enough. I am not a fan of it because it will be very hard to debug: as soon as you call `readout` your state is lost!

I would rather add an `initialize` method to the `MonitoredQuanity` class that would reset the state, that is set the counters to 0, etc.. `initialize` methods of all observed quantities can be called in `NonTheanoBuffer`'s `initialize` method, which in turn would be triggered by `DatasetEvaluator.initialize_aggregators`.

Let me know if you have any questions.